azurerm/helpers/azure/app_service.go

```diff
@@ -1114,16 +1142,55 @@ func FlattenAppServiceLogs(input *web.SiteLogsConfigProperties) []interface{} {
 				blobStorageItem["retention_in_days"] = *blobStorageInput.RetentionInDays
 			}
 
-			blobStorage = append(blobStorage, blobStorageItem)
+			// The API returns a non nil application logs object when other logs are specified so we'll check that this structure is empty before adding it to the statefile.
+			if blobStorageInput.SasURL != nil {
+				blobStorage = append(blobStorage, blobStorageItem)
+
+				appLogsItem["azure_blob_storage"] = blobStorage
+			}
 
-			appLogsItem["azure_blob_storage"] = blobStorage
 		}
 
-		appLogs = append(appLogs, appLogsItem)
+		if len(appLogsItem) > 0 {
+			appLogs = append(appLogs, appLogsItem)
+		}
 
 		result["application_logs"] = appLogs
 	}
 
+	if input.HTTPLogs != nil {
+		httpLogs := make([]interface{}, 0)
+
+		httpLogsItem := make(map[string]interface{})
+
+		if fileSystemInput := input.HTTPLogs.FileSystem; fileSystemInput != nil {
+			fileSystem := make([]interface{}, 0)
+
+			fileSystemItem := make(map[string]interface{})
+
+			if fileSystemInput.RetentionInDays != nil {
+				fileSystemItem["retention_in_days"] = *fileSystemInput.RetentionInDays
+			}
+
+			if fileSystemInput.RetentionInMb != nil {
+				fileSystemItem["retention_in_mb"] = *fileSystemInput.RetentionInMb
+			}
+
+			// The API returns a non nil filesystem logs object when other logs are specified so we'll check that this is disabled before adding it to the statefile.
+			if fileSystemInput.Enabled != nil && *fileSystemInput.Enabled {
+				fileSystem = append(fileSystem, fileSystemItem)
+
+				httpLogsItem["file_system"] = fileSystem
```

Comment:
we need to ensure that `httpLogsItem["file_system"]` is set to an empty list otherwise to ensure the diff show's correctly

Reply:
I set it to an empty list but it caused more issues because it now wants to set `http_logs` even though nothing is present.

Reply:
if there's no elements specified that should match the diff?

Reply:
Got it